src/Microsoft.NetCore.Analyzers/UnitTests/Security/ReviewCodeForSqlInjectionVulnerabilitiesTests.cs

```diff
@@ -0,0 +1,2852 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.NetCore.Analyzers.Security.UnitTests
+{
+    using System;
+    using Microsoft.CodeAnalysis.Diagnostics;
+    using Microsoft.NetCore.Analyzers.Security;
+    using Test.Utilities;
+    using Xunit;
+
+    public class ReviewCodeForSqlInjectionVulnerabilitiesTests : DiagnosticAnalyzerTestBase
+    {
+        protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
+        {
+            return new ReviewCodeForSqlInjectionVulnerabilities();
+        }
+
+        protected override DiagnosticAnalyzer GetBasicDiagnosticAnalyzer()
+        {
+            return new ReviewCodeForSqlInjectionVulnerabilities();
+        }
+
+        protected DiagnosticResult GetCSharpResultAt(int sinkLine, int sinkColumn, int sourceLine, int sourceColumn, string sink, string sinkContainingMethod, string source, string sourceContainingMethod)
+        {
+            this.PrintActualDiagnosticsOnFailure = true;
+            return GetCSharpResultAt(
+                new[] {
+                    Tuple.Create(sinkLine, sinkColumn),
+                    Tuple.Create(sourceLine, sourceColumn)
+                },
+                ReviewCodeForSqlInjectionVulnerabilities.Rule,
+                sink,
+                sinkContainingMethod,
+                source,
+                sourceContainingMethod);
+        }
+
+        protected void VerifyCSharpWithDependencies(string source, params DiagnosticResult[] expected)
+        {
+            this.VerifyCSharp(source, ReferenceFlags.AddTestReferenceAssembly, expected);
+        }
+
+        [Fact]
+        [Trait(Traits.DataflowAnalysis, Traits.Dataflow.TaintedDataAnalysis)]
+        public void HttpRequest_Form_LocalString_Diagnostic()
```

Comment:
>HttpRequest_Form_LocalString_Diagnostic [](start = 20, length = 39)

Can you try this test code with the both CA2100 and your rule being added here enabled? Just want to make sure we don't fire dupes of same issue with different rule ID. If we do, please file a separate tracking issue so we can clean up and remove the overlap with a future PR.

Reply:
#1901